classes/mock/generator.php

```diff
@@ -721,6 +721,7 @@ protected function getParametersSignature(\reflectionMethod $method, $forceMockC
                     $parameterCode .= ' = ' . var_export($parameter->getDefaultValue(), true);
                     break;
 
+                case $parameter->allowsNull():
```

Comment:
nope it will generate

`function f(int $x = null, int $y);`

the issue is that there is no way inside the reflection api to detect the nullable type. This was the only way to make it working (and it's a blocking bug)